Lib/urllib/error.py

```diff
@@ -42,12 +42,10 @@ def __init__(self, url, code, msg, hdrs, fp):
         self.hdrs = hdrs
         self.fp = fp
         self.filename = url
-        # The addinfourl classes depend on fp being a valid file
-        # object.  In some cases, the HTTPError may not have a valid
-        # file object.  If this happens, the simplest workaround is to
-        # not initialize the base classes.
         if fp is not None:
             self.__super_init(fp, hdrs, url, code)
+        else:
+            self.__super_init(io.StringIO(), hdrs, url, code)
```

Comment:
It's both fewer lines and less redundant to do it in one go.

```suggestion
        if fp is None:
            fp = io.StringIO()
        self.__super_init(fp, hdrs, url, code)
```